zephyr/src/utils.c

```diff
@@ -660,6 +660,20 @@ int find_interface_ip(char *ipaddr, int ipaddr_len, char *name) {
 }
 
 int get_mac_address(char *buffer, int size, char *interface) {
+
+    STRUCT_SECTION_FOREACH(net_if, iface)
+    {
+        const struct device *dev = net_if_get_device(iface);
+        if(!strcmp(dev->name, interface)) {
+            sprintf(buffer, "%02x:%02x:%02x:%02x:%02x:%02x",
+                    iface->if_dev->link_addr.addr[0]&0xff, iface->if_dev->link_addr.addr[1]&0xff,
```

Comment:
Looking at this code its obvious you are access internal structures which shouldn't be done as its buggy and can always break, please use the public API `net_if_get_link_addr` to get `net_linkaddr` and then validate that `net_linkaddr` is actually a valid WI-Fi MAC address and then store to `mac[6]` and print.